lib/frameworks/jasmine2.js

```diff
@@ -74,11 +74,29 @@ exports.run = function(runner, specs) {
   var reporter = new RunnerReporter(runner);
   jasmine.getEnv().addReporter(reporter);
 
-  // Filter specs to run based on jasmineNodeOpts.grep and jasmineNodeOpts.invert.
-  jasmine.getEnv().specFilter = function(spec) {
+  // Filter specs/suite to run based on jasmineNodeOpts.grep and jasmineNodeOpts.invert.
+  jasmine.getEnv().specFilter = function(spec, suite) {
+    var suiteMatched = isSuiteMatched(suite);
```

Comment:
This whole section is also a breaking chance.  And what if people want to filter by spec but not suite?